ecmaScriptModule.js

```diff
@@ -0,0 +1,28 @@
+const clownface = require('clownface')
+const iriResolve = require('./lib/iriResolve')
+const ns = require('./namespaces')
+
+async function loader ({ term, dataset }, { basePath } = {}) {
+  const link = clownface({ term, dataset }).out(ns.code('link'))
+
+  if (link.term && link.term.termType !== 'NamedNode') {
+    throw new Error(`Cannot load ecmaScriptModule from term ${term.value}`)
+  }
+
+  const { filename, method } = iriResolve(link.value, basePath)
+
+  const code = await import(filename)
+
+  if (!method) {
+    return code
+  }
+
+  // split method name by . for deep structures
+  return method.split('.').reduce((code, property) => code[property], code)
+}
```

Comment:
This and the `ecmaScript.js` look awfully similar. I would consider combining them

Reply:
I also looked twice, but didn't came up with an easy solution to make one sync the other async. I would like to keep the `require` based implementation backwards compatible.